src/scripting_api/util.js

```diff
@@ -20,11 +20,734 @@ class Util extends PDFObject {
+
+        // Week 1 of a month is the week containing the 3rd
+        // so get the wednesday in week 1.
+        const first = new Date(data.year, data.month, 3);
+        first.setDate(first.getDate() + 2 - ((first.getDay() + 6) % 7));
+
+        return 1 + Math.floor((date - first) / 604800000);
+      },
+      WW: data => {
+        const date = new Date(data.cDate.getTime());
+        // Weeks start on monday.
+        // Date for the closest thursday
+        date.setDate(data.day + 3 - ((data.dayOfWeek + 6) % 7));
+
+        // Week 1 of a year is the week containing January 4
+        // so get the thursday in week 1.
+        const first = new Date(data.year, 0, 4);
+        first.setDate(first.getDate() + 3 - ((first.getDay() + 6) % 7));
+
+        return 1 + Math.floor((date - first) / 604800000);
+      },
+      HH: data => {
+        return data.hours.toString().padStart(2, "0");
+      },
+      H: data => {
+        return data.hours.toString();
+      },
+      hh: data => {
+        return (1 + ((data.hours + 11) % 12)).toString().padStart(2, "0");
+      },
+      h: data => {
+        return (1 + ((data.hours + 11) % 12)).toString();
+      },
+      kk: data => {
+        return (data.hours % 12).toString().padStart(2, "0");
+      },
+      k: data => {
+        return (data.hours % 12).toString();
+      },
+      KK: data => {
+        return (1 + ((data.hours + 23) % 24)).toString().padStart(2, "0");
+      },
+      K: data => {
+        return (1 + ((data.hours + 23) % 24)).toString();
+      },
+      mm: data => {
+        return data.minutes.toString().padStart(2, "0");
+      },
+      m: data => {
+        return data.minutes.toString();
+      },
+      SS: data => {
+        return data.seconds.toString().padStart(2, "0");
+      },
+      S: data => {
+        return data.seconds.toString();
+      },
+      FFF: data => {
+        return data.milliseconds.toString().padStart(3, "0");
+      },
+      A: data => {
+        return data.hours < 12 ? "AM" : "PM";
+      },
+      Z: data => {
+        return data.cDate
+          .toLocaleDateString(undefined, { timeZoneName: "short" })
+          .split(" ")[1];
+      },
+      zz: data => {
+        return this._getTZOffset(data.cDate, ":");
+      },
+      z: data => {
+        return this._getTZOffset(data.cDate, "");
+      },
+    };
+
+    const year = cDate.getFullYear();
+    const month = cDate.getMonth();
+    const day = cDate.getDate();
+    const dayOfWeek = cDate.getDay();
+    const hours = cDate.getHours();
+    const minutes = cDate.getMinutes();
+    const seconds = cDate.getSeconds();
+    const milliseconds = cDate.getMilliseconds();
+    const data = {
+      year,
+      month,
+      day,
+      dayOfWeek,
+      hours,
+      minutes,
+      seconds,
+      milliseconds,
+      cDate,
+    };
+
+    const pattern = /(MMMM|MMM|MM|M|DD|D|YYYY|YY|JJJ|J|EEEE|EEE|E|G|w|WW|hh|h|kk|k|HH|H|KK|K|mm|m|SS|S|FFF|A|Z|zz|z|\\.)/g;
```

Comment:
Which specification did you use for this? I could only find http://citeseerx.ist.psu.edu/viewdoc/download?doi=10.1.1.119.914&rep=rep1&type=pdf#page=842&zoom=auto,-207,487 but there for example the symbol `e` is also allowed and it's not in this regex. I'm not sure if this is the right spec though.

Reply:
It's likely the right spec and mine was older than yours: I added the `e`.
Anyway there is something wrong with this spec: `MM` is used for month and minutes, and a js Date is a date+time... so I don't know how to handle that correctly. So for now I just replaced the `MM` for minutes by `mm`.